document/src/main/java/com/yahoo/document/update/TensorModifyUpdate.java

```diff
@@ -29,8 +30,8 @@ public TensorModifyUpdate(Operation operation, TensorFieldValue tensor) {
     }
 
     private void verifyCompatibleType(TensorType type) {
-        if (type.dimensions().stream().anyMatch(dim -> dim.isIndexed()) ) {
-            throw new IllegalArgumentException("Tensor type '" + type + "' is not compatible as it contains some indexed dimensions");
+        if (type.rank() > 0 && type.dimensions().stream().noneMatch(dim -> dim.isMapped()) ) {
```

Comment:
This will break how we handle such updates in backend. We treat the cells to update as a sparse tensor and assumes this has been verified before serialized. To support the blocked syntax, the resulting tensor must still be a sparse one. Also see tensor_modify_update_treats_the_input_tensor_as_sparse().

Reply:
Okay, let's convert in this layer for now https://github.com/vespa-engine/vespa/pull/11766